multiple-servers/api/api.go

```diff
@@ -0,0 +1,134 @@
+package api
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+	"multiple-servers/api/images"
+	. "multiple-servers/api/types"
+	"net/http"
+	"strings"
+
+	"github.com/jackc/pgx/v4"
+)
+
+type Config struct {
+	Port   int    `json:"path"`
+	DB_URL string `json:"port"`
+}
+
+type Server struct {
+	db *pgx.Conn
+}
+
+func Run(c Config) error {
+	conn, err := pgx.Connect(context.Background(), c.DB_URL)
+	if err != nil {
+		return err
+	}
+	defer conn.Close(context.Background())
+
+	s := Server{db: conn}
+
+	http.HandleFunc("/", s.IndexHandler)
+	http.HandleFunc("/images.json", s.ImagesHandler)
+
+	err = http.ListenAndServe(fmt.Sprintf(":%d", c.Port), nil)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (s *Server) IndexHandler(w http.ResponseWriter, r *http.Request) {
+	log.Println(r.Method, r.URL.EscapedPath())
+	enableCors(&w)
+
+	switch r.Method {
+	case "GET":
+		w.Write([]byte("Welcome to the API"))
+	default:
+		w.WriteHeader(http.StatusMethodNotAllowed)
+	}
+}
+
+func (s *Server) ImagesHandler(w http.ResponseWriter, r *http.Request) {
+	log.Println(r.Method, r.URL.EscapedPath())
+	enableCors(&w)
+
+	indent := 1
+	queryParams := r.URL.Query()
+	if queryParams.Get("indent") != "" {
+		i, err := ValidateIndent(queryParams.Get("indent"))
+		if err != nil {
+			log.Println(err)
+			w.WriteHeader(http.StatusBadRequest)
+			w.Write([]byte(err.Error()))
```

Comment:
I decided to not test every scenario in HTTP since they are already tested in unit tests. My reasoning - if functionality for the function responsible for indent is updated, then tests in both places need to be updated even though the change might not affect HTTP functionality as much.

Reply:
Absolutely makes sense, good reasoning!